### DISCUSSION TITLE
Dropzone in form operation? How?

### DISCUSSION THREAD

**Discussion by kde99** [Laravel-Backpack/community-forum#730] [Q&A (Help)] [answered]
Hello everyone!

I am creating a ticketing subsystem into our CRM and I want to create a form operation that would basically let the user quickly edit a ticket event. I created a form operation which has 2 fields. A text and a dropzone. The controller is built like ` TicketEventCRUD (this has the DropzoneOperation trait) -extends> MyTicketEventsCRUD -trait> QuickEditFormOperationTrait `

The operation defines the form in the getQuickEditForm function:
```php
public function getQuickEditForm(?int $id = null)
{
        $this->crud->hasAccessOrFail('quickEdit');

        $event = TicketEvents::findOrFail($id);

        $this->crud->field("result")->type("wysiwyg")->default($event->result);
        $this->crud->field("attachments")
            ->type("dropzone")
            ->default($event->attachments)
            ->withFiles([
                "path" => "tickets/" . date("Y/m"),
            ]);

        return $this->formView($id);
}
```

This shows the field correctly. When I try to upload a file it just shows "[Object object]" under the field and the network console shows that an error is sent back from the upload handler "dropzone/upload" with the message of `Cannot parse field name.`. 

Looking at the stacktrace it shows that the error is thrown at `backpack\\pro\\src\\Http\\Controllers\\Operations\\DropzoneOperation.php at line 141` which is the getDropzoneFieldName() function, which calls for the isValidDropzoneField() function which return false for my dropzone field, as the `$this->crud->getFields()` returns an empty array.

My question:
1. Is this a limitation of the form operation?
2. Can I somehow bypass this error and make use of dropzones in my operation?

Thanks for any help!

**Comment by phpfour** [comment, marked as answer]
Hey @kde99, thanks for your question. 

I'm afraid the `getQuickEditForm()` method is not the right place for adding the fields. The documentation mentions two different places for defining them: 

![CleanShot 2023-11-06 at 08 19 58@2x](https://github.com/Laravel-Backpack/community-forum/assets/171715/168aefd8-e9cf-4cab-88b9-1bbd719fbc32)

In your case, as you want to set some defaults from the main entity, the first option is suitable for you where you define a `setupQuickEditOperation()` method in your Crud Controller like below:

```php
public function setupQuickEditOperation()
{
    $event = $this->crud->getCurrentEntry();

    $this->crud->field("result")
        ->type("wysiwyg")
        ->default($event->result);
    
    $this->crud->field("attachments")
        ->type("dropzone")
        ->default($event->attachments)
        ->withFiles([
            "path" => "tickets/" . date("Y/m"),
        ]);
}
```

Hope this resolves the issue for you. 

Cheers!

**Reply by kde99** [reply]
Thank you! I did read that part of the docs, but I didn't know what it really meant, but now it's all clean.
